### ISSUE TITLE
Neovide does not seem to be picking up zsh configuration from non-standard location 

### ISSUE THREAD

**Issue by MostHated** [neovide/neovide#1773]
**Describe the bug**
My .zsh configuration is in a non-standard location. While I do have a ~/.zshenv in the standard location, and I am manually setting `ZDOTDIR`= It doesn't seem to be picking things up properly.

**Expected behavior**
nvm, and in turn, my other npm settings for version usage, etc which are set in my profile, would be respected.

**Screenshots**
![](https://img.instance.id/2z3EV6PRdpvx/direct)

**Desktop (please complete the following information):**
 - OS: Pop_OS 22.04
 - Neovide Version : `neovide 0.10.3`
 - Neovim Version : ` v0.9.0-dev-554+g6f25623e7`

**Please run `neovide --log` and paste the contents of the `.log` file created in the current directory here:**
https://gist.github.com/MostHated/6308248793d549bdeb23b405a32016d1

**Additional context**
I have the following set in my .zshenv file, and then there is ~/.zsh/.zshrc in there along with everything else related.

```sh
...
export ZDOTDIR="$HOME/.zsh"
...
```

What is recommended to solve this, a wrapper script in which I set vars ahead of time, or is there something native that can be configured?

**Comment by xbladesub** [comment]
same here

**Comment by jonlinnell** [comment]
Launching neovide from a terminal causes it to pick up the correct environment.

I'd prefer not to have to do this (this is one of the reasons I don't use Neovide more regularly) but it's an effective workaround.

**Comment by MostHated** [comment]
Yeah, I created a wrapper script and use that for a keybinding. As you mentioned, I would rather not have to if possible.

**Comment by luizcarlos1405** [comment]
I set the option `shell = "zsh"` and now it sources `.zshrc`. Like so:

`vim.o.shell = "zsh`

**Comment by luizcarlos1405** [comment]
I needed the `.zshrc` file to be sourced because of changes to the `PATH`. So I added this to my Neovide config to fix it:

`vim.env.PATH = "/my/folder/path/bin:" .. vim.env.PATH`

Easy to add any folder to the `PATH`. Don't forget the `:` at the end.

**Comment by fredizzimo** [comment]
This is actually part of the FAQ https://neovide.dev/faq.html#neovide-is-not-picking-up-some-shell-configured-information

Neovide does not start an interactive shell to run Neovim inside, which is a requirement for sourcing the `.zshrc` file. On WSL and Mac outsde the teriminal it starts a login shell, but on Linux it does not start a shell at all, but your desktop environment probably loads the `.zprofile` so you should be able to put  the things you need there.

**Comment by TheTharin** [comment]
> Neovide does not start an interactive shell to run Neovim inside, which is a requirement for sourcing the `.zshrc` file. On WSL and Mac outsde the teriminal it starts a login shell

Mac Sonoma 14.5 M2
Doesn't look like it does. PATH is set up correctly to have `/opt/homebrew/bin/nvim`
If installed directly from the DMG and ran - `zsh: command not found nvim`

**Comment by fredizzimo** [comment]
@TheTharin, in which file do you setup homebrew? (`eval "$(/opt/homebrew/bin/brew shellenv)"`)? It should be in `.zprofile` if your default shell is zsh.

**Comment by TheTharin** [comment]
> @TheTharin, in which file do you setup homebrew? (`eval "$(/opt/homebrew/bin/brew shellenv)"`)? It should be in `.zprofile` if your default shell is zsh.

Adding this helped! Thanks!
